### PR TITLE
The user can reset the password 

### PR DESCRIPTION
**Pull request by gracelungu**
#### What does this PR do?
Enables the user to reset the password
#### Description of Task to be completed?
~ Write the endpoint for resetting the password
~ Use Nodemailer to send the email
~ Handle request errors
~ Write tests for the reset password endpoint
#### How should this be manually tested?
This can be manually tested on Heroku using postman
#### Any background context you want to provide?
This was the last endpoint for the challenge 3
#### What are the relevant pivotal tracker stories?
164034661

### CODE REVIEW

File: server/controllers/user.js [hunk 1]
```diff
@@ -222,6 +277,20 @@ const createCandidate = async (req, res) => {
 };
 
 const vote = async (req, res) => {
+  const voteSchema = schema({
```
Comment (dotNesh) [inline server/controllers/user.js:280]: What does the vote schema have to do with a password reset?

**Comment by dotNesh** [comment]
@gracelungu For your tests, apart from testing the status code ensure you also test the exact error/success message the user is supposed to get. For example `Bad request` or a value on the JSON they receive.

**Review by dotNesh — CHANGES_REQUESTED** [review]
Please addressc the requested changes.

File: server/controllers/user.js (hunk 1 above)
Comment (gracelungu) [reply server/controllers/user.js:280]: While working on the reset password feature, I also made some changes to other controllers.
Is it a bad practice? Should I only make changes on the specific things I am working on?

**Comment by gracelungu** [comment]
About the json data response, should i validate all the data object fields? Like the checking if the data.email is a valide email adress inside the test?.

**Pull request merged**